tests/unit/test_charm.py

```diff
@@ -181,6 +240,168 @@ def test_on_config_changed_iscsi(harness, mocker, iscsi_config):
     assert isinstance(harness.charm.unit.status, ActiveStatus)
 
 
+def test_iscsi_with_initiatorname_rendered(mocker, harness, iscsi_config):
+    """Test iscsi configuration when initiatorname.iscsi is present and rendered again.
+
+    The test is performed when the initiator name from initiatorname.iscsi contains
+    a different value from the one in the initiator-dictionary.
+    """
+    mocker.patch("charm.utils.is_container", return_value=False)
+    mocker.patch("charm.subprocess.getoutput", return_value="iqn.2020-07.canonical.com:lun1")
+    mocker.patch("charm.socket.getfqdn", return_value="testhost.testdomain")
+    mocker.patch("charm.subprocess.check_call")
+    mocker.patch("charm.subprocess.check_output")
+    mocker.patch("charm.StorageConnectorCharm._configure_deferred_restarts")
+    mocker.patch("charm.StorageConnectorCharm._defer_service_restart")
+    mocker.patch("charm.Path.chmod")
+
+    # configure the initiatorname.iscsi file with a different name from the one in
+    # the initiator-dictionary. The file should be rendered again from the name in
+    # the dictionary
+    initiator_content_with_different_name = dedent(
+        """\
+        ###############################################################################
+        # [ WARNING ]
+        # configuration file maintained by Juju
+        # local changes will be overwritten.
+        #
+        # DO NOT EDIT OR REMOVE THIS FILE!
+        # If you remove this file, the iSCSI daemon will not start.
+        # If you change the InitiatorName, existing access control lists
+        # may reject this initiator.  The InitiatorName must be unique
+        # for each iSCSI initiator.  Do NOT duplicate iSCSI InitiatorNames.
+        ###############################################################################
+        InitiatorName=new-iqn"""
+    )
+    harness.charm.ISCSI_INITIATOR_NAME.write_text(initiator_content_with_different_name)
+    with open(harness.charm.ISCSI_INITIATOR_NAME) as f:
+        content = f.read()
+    assert "InitiatorName=new-iqn" in content
```

Comment:
I think we don't need to validate this one, because the content is written by unit test, not by the source code itself.